src/components/onboarding/steps/collector_intent/index.tsx

```diff
@@ -0,0 +1,77 @@
+import * as React from "react"
+import styled from "styled-components"
+
+import Step from "../step"
+import Option from "./option"
+
+const OptionsContainer = styled.div`
+  width: 450px;
+  margin: 0 auto 100px;
+`
+
+interface Props {
+  onStateChange?: ({ nextButtonEnabled }) => void
+}
+
+interface State {
+  selectedOptions: { [option: string]: boolean }
+  selectedCount: number
+}
+
+class CollectorIntent extends React.Component<Props, State> {
+  options = [
+    "Buy Art & Design",
+    "Sell Art & Design",
+    "Research Art Prices",
+    "Learn About Art",
+    "Find Out About New Exhibitions",
+    "Read Art Market News",
+  ]
+
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      selectedOptions: {},
+      selectedCount: 0,
+    }
+  }
```

Comment:
I always call this out, but there's no need to use constructors in TypeScript because ES7 class properties are supported out of the box:

```js
class Foo extends Component {
  state = {
    selectedOptions: {},
    selectedCount: 0,
  }

  ... 
}

Reply:
Thanks for pointing that out @damassi. FWIW totally aware of ES7 class properties support in TS and maybe an unfounded concern of mine but I tend to sway towards staying consistent throughout to avoid confusion as there are components that need to initialize their state based on props in the project. I'm also a fan of less boilerplate so it's a good idea to just start using it.